numpy/core/tests/test_conversion_utils.py

```diff
@@ -7,19 +7,31 @@
 
 import numpy as np
 import numpy.core._multiarray_tests as mt
+from numpy.testing import assert_warns
 
 
 class StringConverterTestCase:
     allow_bytes = True
     case_insensitive = True
     exact_match = False
+    warn = True
 
     def _check_value_error(self, val):
         pattern = r'\(got {}\)'.format(re.escape(repr(val)))
         with pytest.raises(ValueError, match=pattern) as exc:
             self.conv(val)
 
+    def _check_conv_assert_warn(self, val, expected):
+        if self.warn:
+            with assert_warns(DeprecationWarning) as exc:
+                assert self.conv(val) == expected
```

Comment:
Since this is a C api, it would be good to try making warnings errors, and catching that here too.

Reply:
I think this is covered by the tests in `numpy/core/tests/test_deprecations.py` but it surfaces as ValueError instead of DeprecationWarning. I will look into why string_converter_helper does this.

Reply:
since `numpy/core/tests/test_deprecations.py` already covers raising warnings as errors, can we keep test_conversion_utils as it is ?

Reply:
Sure, that's reasonable.